apps/web/src/components/templates/email-editor/EmailContentCard.tsx

```diff
@@ -22,14 +25,25 @@ export function EmailContentCard({
   isIntegrationActive: boolean;
 }) {
   const { readonly } = useEnvController();
+  const theme = useMantineTheme();
   const {
     control,
     formState: { errors },
     setValue,
     watch,
   } = useFormContext(); // retrieve all hook methods
   const contentType = watch(`steps.${index}.template.contentType`);
+  const subject = watch(`steps.${index}.template.subject`);
   const [activeTab, setActiveTab] = useState(0);
+  const { integrations = [] } = useIntegrations();
+  const [integration, setIntegration]: any = useState(null);
+
+  useEffect(() => {
+    if (integrations.length === 0) {
+      return;
+    }
+    setIntegration(integrations.find((item) => item.channel === 'email') || null);
+  }, [integrations]);
```

Comment:
```suggestion
  }, [integrations, setIntegration]);
```
When having the linter set for hooks it will complain about this.